plugins/msg-broker/mcollective/lib/openshift/mcollective_application_container_proxy.rb

```diff
@@ -3330,8 +3332,17 @@ def self.select_best_fit_node_impl(server_infos)
 
       server_info = nil
       unless server_infos.empty?
-        # Randomly pick one of the best options
-        server_info = server_infos[rand(server_infos.length)]
+        server_infos.sort_by! { |server_info| server_info.node_consumed_capacity }.reverse!
+        # Weight the servers by their availability
+        # Divide by 2 gives a 3:1 ratio of most to least available selection.  Divide by 4 would give 5:1.
+        weight_skew = (server_infos.length.to_f / 2).round
+        # The most available nodes are at the end of the list and are associated with a larger portion of the sum
+        weights_sum = (((server_infos.length.to_f/2) * (server_infos.length-1)) + (weight_skew * server_infos.length)).to_i
+        random_weighted_position = rand(weights_sum)
+
+        # Map the random weighted position into its corresponding index
+        random_index = (0.5 - weight_skew + Math.sqrt((weight_skew - 0.5)**2 + (2 * random_weighted_position))).floor
```

Comment:
Should work fine... but given how long it took me to determine that, I would worry about support cases coming in with people having to be convinced of that, versus the simply random selection.